### PR TITLE
[#ICC-167]  Fix get message error 500 when retrieving message without content

### PR DESCRIPTION
**Pull request by fabriziopapi**
The getContentFromBlob method of Message model do not set apart empty blob from not existing blob. A not exisitng blob should not throw and error, becouse a message without content is valid.

#### List of Changes

add new getBlobAsTextWithError to azure utils
change getContentFromBlob method to manage 404 get blob error

#### Motivation and Context

The message model is used to retrieve a message. A Service poll this method via fn-service API to known the status message as part of send message process. This polling must return a message without content if the message is not rejected/refused, but not yet processed.

#### How Has This Been Tested?

unit test

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/utils/azure_storage.ts [hunk 1]
```diff
@@ -186,6 +187,35 @@ export const getBlobAsText = (
...
   });
 
+/**
+ * Get a blob content as text (string). In case of error, it will return error details.
+ *
+ * @param blobService     the Azure blob service
+ * @param containerName   the name of the Azure blob storage container
+ * @param blobName        blob file name
+ */
+export const getBlobAsTextWithError = (
```
Comment (gquadrati) [inline src/utils/azure_storage.ts:197]: How did you test it? Maybe it's better to add an integration test

**Review by gquadrati — CHANGES_REQUESTED** [review]

File: src/utils/azure_storage.ts (hunk 1 above)
Comment (fabriziopapi) [reply src/utils/azure_storage.ts:197]: added for whole getContentBlob

File: src/models/__tests__/message.test.ts [hunk 2]
```diff
@@ -790,11 +792,36 @@ describe("getContentFromBlob", () => {
...
+        "The specified blob does not exist.\nRequestId:27149c2d-001e-000e-2d04-a12766000000\nTime:2022-07-26T15:27:59.0934919Z",
+      code: "BlobNotFound",
+      statusCode: 404,
+      requestId: "27149c2d-001e-000e-2d04-a12766000000"
+    };
     const getBlobAsTextSpy = jest
-      .spyOn(azureStorageUtils, "getBlobAsText")
-      .mockReturnValueOnce(Promise.resolve(E.left(err)));
+      .spyOn(azureStorageUtils, "getBlobAsTextWithError")
+      .mockReturnValueOnce(() => TE.left(err));
```
Comment (gquadrati) [inline src/models/__tests__/message.test.ts:806]: We could add another test with error having a different error code

File: src/utils/azure_storage.ts [hunk 3]
```diff
@@ -186,6 +187,35 @@ export const getBlobAsText = (
...
+        blobService.getBlobToText(
+          containerName,
+          blobName,
+          options,
+          (err, result, _) =>
+            err ? resolve(E.left(err)) : resolve(E.right(fromNullable(result)))
+        );
+      }
+    ),
+    constant
```
Comment (gquadrati) [inline src/utils/azure_storage.ts:216]: Why not using `TE.tryCatch` here?

File: src/utils/__tests__/azure_storage.test.ts [hunk 4]
```diff
@@ -8,9 +8,12 @@ import { isNone } from "fp-ts/lib/Option";
 import {
   BlobNotFoundCode,
   getBlobAsObject,
-  getBlobAsText
+  getBlobAsText,
+  getBlobAsTextWithError
```
Comment (gquadrati) [inline src/utils/__tests__/azure_storage.test.ts:12]: Why importing it, since it's not used? Is there a non-committed test?

**Review by gquadrati — APPROVED** [review]
LGTM, just some small comments

File: src/utils/azure_storage.ts (hunk 3 above)
Comment (fabriziopapi) [reply src/utils/azure_storage.ts:216]: The tryCatch will route a throwed error in the left side. In this case we had a callback managing the error.
How could we use the tryCatch to manage it?

PS: I tried with TE.taskify, but I could not understand how to make it work.

File: src/models/__tests__/message.test.ts (hunk 2 above)
Comment (fabriziopapi) [reply src/models/__tests__/message.test.ts:806]: done. thank you!

File: src/utils/__tests__/azure_storage.test.ts (hunk 4 above)
Comment (fabriziopapi) [reply src/utils/__tests__/azure_storage.test.ts:12]: no, but I'm adding some test. thank you!

**Pull request merged**